worlds/tmc/Rules.py

```diff
@@ -778,8 +778,7 @@ def __init__(self, world: "MinishCapWorld") -> None:
                                                self.logic_and([self.has_all([TMCItem.GUST_JAR, TMCItem.LANTERN]),
                                                                self.has(TMCItem.SMALL_KEY_DWS, 1)])])]),
             TMCLocation.DEEPWOOD_1F_MADDERPILLAR_HP:
-                self.logic_or([self.has(TMCItem.SMALL_KEY_DWS, 4),
-                               self.logic_and([self.has(TMCItem.SMALL_KEY_DWS, 2), self.has(TMCItem.LANTERN)]),
+                self.logic_or([self.logic_and([self.has(TMCItem.SMALL_KEY_DWS, 2), self.has(TMCItem.LANTERN)]),
                                self.logic_and([self.has_all([TMCItem.GUST_JAR, TMCItem.LANTERN]),
                                                self.has(TMCItem.SMALL_KEY_DWS, 1)])]),
```

Comment:
This doesn't account for having 4 keys and no lantern. Shouldn't it still check for gust jar and the 4 keys?
```suggestion
                self.logic_or([self.logic_and([self.has(TMCItem.SMALL_KEY_DWS, 4), self.has(TMCItem.GUST_JAR)]),
                               self.logic_and([self.has(TMCItem.SMALL_KEY_DWS, 2), self.has(TMCItem.LANTERN)]),
                               self.logic_and([self.has_all([TMCItem.GUST_JAR, TMCItem.LANTERN]),
                                               self.has(TMCItem.SMALL_KEY_DWS, 1)])]),
```